test/app-luatest/gh_7808_rel_paths_in_diag_and_debug_info_test.lua

```diff
@@ -0,0 +1,25 @@
+local bit = require('bit')
+local misc = require('test.luatest_helpers.misc')
+local netbox = require('net.box')
+local t = require('luatest')
+
+local g = t.group()
+
+g.test_rel_paths_in_diag = function(_)
+    local _, err = pcall(function() netbox.self:call('') end)
+    t.assert_not(err.trace[1].file:startswith('/'))
```

Comment:
Will `box.error.new(box.error.UNKNOWN).trace[1].file` suit you here?

Reply:
No, such a trace will show the test file, I need a diagnostic set from inside a source file.